component.js

```diff
@@ -121,24 +120,43 @@ define([
 					? node[CHILDREN]
 					: node[CHILDREN] = [];
 
-				return when.unfold(function (index) {
+				var seed = 0;
+
+				function unspool (index) {
 					var child;
 
 					// Find next child without a `COMPLETED` property
 					do {
-						child = children[ index++ ];
+						child = children[index++];
 					}
 					while (child !== UNDEFINED && child.hasOwnProperty(COMPLETED));
 
-					return [ child, index ];
-				}, function (index) {
+					return [child, index];
+				}
+
+				function predicate (index) {
 					// Check if we're out of bounds. Note that we allow _adding_ to `children` during `unfold`
 					return index >= children[LENGTH];
-				}, function (child) {
-					if (child !== UNDEFINED) {
+				}
+
+				function handler (child) {
+					//if (child !== UNDEFINED && child[COMPONENT] !== UNDEFINED) {
+					if (typeof child !== UNDEFINED && child[COMPONENT] !== UNDEFINED) {
 						return ready.call(child[COMPONENT]());
 					}
-				}, 0);
+				}
+
+				return when.unfold(unspool, predicate, handler, seed).catch(function(error)
+				{
+					if(error && error instanceof Error)
+					{
+						console.log('troop-contrib-com component yield error message:: '  +  error.message + ' node:: ' + node +' stack:: ' + error.stack + ' ' );
```

Comment:
Will do as part of the raygun ticket want to verify the problem is resolved
first.


On Thu, Mar 24, 2016 at 1:38 PM, davlou <notifications@github.com> wrote:

> In component.js
> <https://github.com/troopjs-contrib/troopjs-contrib-com/pull/2#discussion_r57316766>
> :
>
> >  						return ready.call(child[COMPONENT]());
> >  					}
> > -				}, 0);
> > +				}
> > +
> > +				return when.unfold(unspool, predicate, handler, seed).catch(function(error)
> > +				{
> > +					if(error && error instanceof Error)
> > +					{
> > +						console.log('troop-contrib-com component yield error message:: '  +  error.message + ' node:: ' + node +' stack:: ' + error.stack + ' ' );
>
> Can we push these to the logs (Raygun?) instead of the console?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/troopjs-contrib/troopjs-contrib-com/pull/2/files/f6d3a0c050fa08dbebdc5f72969433d6ae0f460f#r57316766>
>